recipes/full_finetune_distributed.py

```diff
@@ -222,6 +224,7 @@ def setup(self, cfg: DictConfig) -> None:
 
         self._optimizer = self._setup_optimizer(
             cfg_optimizer=cfg.optimizer,
+            optimizer_in_bwd=self._optimizer_in_bwd,
```

Comment:
should `_optimizer_in_bwd` be part of `cfg.optimizer`, or we did so to mimic single device recipe ?

Reply:
I follow the structure from [single_device](https://github.com/pytorch/torchtune/blob/4bbed4d4e245e990208f3389689291622a7af9df/recipes/full_finetune_single_device.py#L225), shall we restructure both of the distributed and single_deivce?